aea/contracts/base.py

```diff
@@ -18,44 +18,53 @@
 # ------------------------------------------------------------------------------
 
 """The base contract."""
-
-from abc import ABC
-from typing import Any, Dict
+import json
+from pathlib import Path
+from typing import Any, Dict, Optional
 
 from aea.configurations.base import ContractConfig, ContractId
+from aea.configurations.components import Component
 
 
-class Contract(ABC):
+class Contract(Component):
     """Abstract definition of a contract."""
 
     def __init__(
-        self,
-        contract_id: ContractId,
-        config: ContractConfig,
-        contract_interface: Dict[str, Any],
+        self, config: ContractConfig,
     ):
         """
         Initialize the contract.
 
-        :param contract_id: the contract id.
         :param config: the contract configurations.
-        :param contract_interface: the contract interface
         """
-        self._contract_id = contract_id
-        self._config = config
-        self._contract_interface = contract_interface
+        super().__init__(config)
+        self._contract_interface = None  # type: Optional[Dict[str, Any]]
 
     @property
     def id(self) -> ContractId:
         """Get the name."""
-        return self._contract_id
+        return self.public_id
 
+    # TODO to remove
```

Comment:
?

Reply:
Contract inherits `configuration` from parent class `Component`, hence `config` is obsolete.